packages/kbn-reporting/public/share/share_context_menu/image_export_modal.tsx

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  EuiButton,
+  EuiIcon,
+  EuiButtonEmpty,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiModalFooter,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
+  EuiToolTip,
+  EuiText,
+} from '@elastic/eui';
+import type { IUiSettingsClient, ThemeServiceSetup, ToastsSetup } from '@kbn/core/public';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n-react';
+import url from 'url';
+import React, { FC, useCallback, useEffect, useState, useMemo } from 'react';
+import useMountedState from 'react-use/lib/useMountedState';
+import { LayoutParams } from '@kbn/screenshotting-plugin/common';
+import type { JobAppParamsPDFV2 } from '@kbn/reporting-export-types-pdf-common';
+import { BaseParams } from '@kbn/reporting-common/types';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+import { ReportingAPIClient } from '../..';
+import { JobParamsProviderOptions } from '.';
+import { ErrorUrlTooLongPanel } from './reporting_panel_content/components';
+import { getMaxUrlLength } from './reporting_panel_content/constants';
+
+export interface ReportingModalProps {
+  apiClient: ReportingAPIClient;
+  toasts: ToastsSetup;
+  uiSettings: IUiSettingsClient;
+  reportType?: string;
+  requiresSavedState: boolean; // Whether the report to be generated requires saved state that is not captured in the URL submitted to the report generator.
+  objectId?: string;
+  isDirty?: boolean;
+  onClose: () => void;
+  theme: ThemeServiceSetup;
+  layoutOption?: 'print' | 'canvas';
+  // canvas breaks if required
+  jobProviderOptions?: JobParamsProviderOptions;
+  // needed for canvas
+  getJobParams?: JobAppParamsPDFV2;
+  objectType: string;
+  downloadCsvFromLens: () => void;
+}
+
+type AppParams = Omit<BaseParams, 'browserTimezone' | 'version'>;
+
+export type Props = ReportingModalProps & { intl: InjectedIntl };
+
+type AllowedImageExportType = 'pngV2' | 'printablePdfV2' | 'printablePdf' | 'csv';
```

Comment:
* `csv` isn't an image type
*`printablePdf` is deprecated, so is only available via API, not the UI

```suggestion
type AllowedImageExportType = 'pngV2' | 'printablePdfV2';
```